test/Makefile.am

```diff
@@ -9,9 +9,11 @@
 SUBDIRS = mpi commands .
 
 # Test both the MPI routines and the MPICH command scripts
-testing:
+testing: mpi/Makefile
 	(NOXMLCLOSE=YES && export NOXMLCLOSE && cd mpi && $(MAKE) testing)
 	(XMLFILE=../mpi/summary.xml && XMLCONTINUE=YES && \
 	export XMLFILE && export XMLCONTINUE && \
 	cd commands && $(MAKE) testing)
 
+mpi/Makefile:
+	cd test && ./autogen.sh && ./configure
```

Comment:
Does this depend on the MPI being available in the user `PATH` then?

Reply:
It does. The configure will fail if `mpicc` can't be found, and thus `make testing` will fail. Does the testing work currently without setting path? If it does, I guess we just need pass the prefix to `test/configure` so it can find `mpicc` and `mpiexec`.

Reply:
Yes, it currently works without having to set anything extra. It is able to inherit the install prefix from the MPICH configure.

Reply:
Direct `make testing` without setting paths should work now.